### DISCUSSION TITLE
Axios Response Intercepter not passing response data, Vuejs

### DISCUSSION THREAD

**Discussion by bribar** [axios/axios#5829] [Q&A]
```
axios.interceptors.response.use(
(response) => {
	
    return response

}, async (error) => {

const instance = axios.create({
	baseURL: 'myrestbase',
	timeout: 1000,
	headers: {
		'Authorization': 'Bearer TOKEN HERE',
		'Accept' : 'application/json',
		'Content-Type' : 'application/json'
	}
})

if(error.response.status == 401){
	await instance.post('/token/refresh').then(tokenResponse => {
		
		if(tokenResponse.data.status){

			error.config.headers['Authorization'] = 'Bearer ' + tokenResponse.data.raw_token
			
			return axios(error.config)
			// router.go(router.currentRoute)

		}else{
			
			ElMessage({
				message: 'Your session has expired! Please login again.',
				type: 'warning'
			})

			logout()

		}

	}).catch(error => {
	
		logout()

	})

} 

})
```
My interceptor seems to work as expected. If a request gets a 401 response (token expired), I refresh the token. On success, for the new token, I want to complete the original request with the updated token, which I'm attempting with `axios(error.config)`. This works fine...but the original request response comes up `undefined'.

```
axios.get('/original_request', {
    params: payload
}).then(response => {
    
    console.log(response)

})
```
I can get by with `router.go(router.currentRoute)` that I have commented out, and most won't recognize a route reload in Vuejs. I would really like it to be seamless, especially if it's a POST request where the user input form fields, etc.

Anyone have any suggestions how I can get the data in the response instead of `undefined`?

**Comment by jasonsaayman** [comment]
The retry response is `undefined` because the error handler does not return the promise from the refresh/retry chain.

Inside this block:

```js
await instance.post('/token/refresh').then(...)
```

the `return axios(error.config)` only returns from the inner `.then()` callback, not from the interceptor's error handler. The interceptor itself then resolves with `undefined`.

Return the retry promise from the interceptor:

```js
axios.interceptors.response.use(
  response => response,
  async error => {
    if (error.response?.status !== 401) {
      return Promise.reject(error);
    }

    try {
      const tokenResponse = await instance.post('/token/refresh');

      if (!tokenResponse.data.status) {
        logout();
        return Promise.reject(error);
      }

      error.config.headers.Authorization = `Bearer ${tokenResponse.data.raw_token}`;

      return axios(error.config);
    } catch (refreshError) {
      logout();
      return Promise.reject(refreshError);
    }
  }
);
```

Also consider using an `_retry` flag on `error.config` to avoid an infinite loop if the retried request also returns `401`.
